code/game/objects/machinery/bots/floorbot.dm

```diff
@@ -317,34 +317,19 @@
 	else
 		src.icon_state = "floorbot[src.on]e"
 
-/obj/machinery/bot/floorbot/explode()
-	src.on = 0
-	src.visible_message("<span class='danger'>[src] blows apart!</span>", 1)
-	var/turf/Tsec = get_turf(src)
-
-	var/obj/item/storage/toolbox/mechanical/N = new /obj/item/storage/toolbox/mechanical(Tsec)
-	N.contents = list()
-
-	new /obj/item/assembly/prox_sensor(Tsec)
-
-	if (prob(50))
-		new /obj/item/robot_parts/l_arm(Tsec)
-
-	while (amount)//Dumps the tiles into the appropriate sized stacks
-		if(amount >= 16)
-			var/obj/item/stack/tile/plasteel/T = new (Tsec)
-			T.amount = 16
-			amount -= 16
-		else
-			var/obj/item/stack/tile/plasteel/T = new (Tsec)
-			T.amount = src.amount
-			amount = 0
+/obj/machinery/bot/floorbot/deconstruct(disassembled = TRUE)
+	new /obj/item/storage/toolbox/mechanical(loc)
```

Comment:
shouldn't these be in the `component_parts` list?

Reply:
No these are for the actual stock_parts